### ISSUE TITLE
Better waiting for chrome instance to launch

### ISSUE THREAD

**Issue by andrewvy** [andrewvy/chrome-launcher#8]
Instead of polling blindly right when we launch the chrome process with erl-exec, a hacky-way is to wait for stdout before polling.

**Comment by andrewvy** [comment]
Looks like there is already discussion of "pre-opening a socket" before `exec` a chrome instance, and then passing it in as an option at startup: https://bugs.chromium.org/p/chromium/issues/detail?id=624837

Though, I'm not sure if [erlexec](https://github.com/saleyn/erlexec) is able to handle this.

Additional docs: https://unix.stackexchange.com/questions/248408/file-descriptors-across-exec

---

Actually, there's a hackier workaround if ^ doesn't work. ([Comment #4](https://bugs.chromium.org/p/chromium/issues/detail?id=624837#c4))

**Comment by andrewvy** [comment]
Some options:

- Wait for stdout message that has a format of `DevTools listening on 127.0.0.1:9222`, then poll. (This is what [puppeteer](https://github.com/GoogleChrome/puppeteer/blob/master/lib/Launcher.js#L196) does, not sure why it's a separate implementation from chrome-launcher)

- Wait for `DevToolsActivePort` file to get created in `--user-data-dir`, then poll.

- Figure out how to pass file descriptors // pre-open socket and pass it into chrome

**Comment by bcardarella** [comment]
> Wait for stdout message that has a format of

I don't think this is a viable option. Headless outputs *everything* to stderr. I logged a bug about this last week and it was marked as `wontfix`

https://bugs.chromium.org/p/chromium/issues/detail?id=773572&can=1&q=stderr&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified

**Comment by andrewvy** [comment]
Argh, so I was trying to naively implement Option 1, but there was a couple things that made it not as straightforward to implement.

Chrome Launcher currently doesn't do any wrapping, it links the caller process with the started `:exec.run_link` process that gets created. Not GenServer-wrapped. To implement this functionality, we'd have to filter through the stderr log messages and send a message.
